src/migrations/data.ts

```diff
@@ -508,6 +510,44 @@ function fixChartDefinitions(data: Partial<WorkbookData>, initialMessages: State
   return messages;
 }
 
+function fixTranslatedDuplicateSheetName(
+  data: Partial<WorkbookData>,
+  initialMessages: StateUpdateMessage[]
+): StateUpdateMessage[] {
+  const sheetNames = {};
+  for (const sheet of data.sheets || []) {
+    sheetNames[sheet.id] = sheet.name;
+  }
+  const messages: StateUpdateMessage[] = [];
+  for (const message of initialMessages) {
+    if (message.type === "REMOTE_REVISION") {
+      const commands: CoreCommand[] = [];
+      for (const cmd of message.commands) {
+        switch (cmd.type) {
+          case "DUPLICATE_SHEET":
+            cmd.sheetNameTo = getDuplicateSheetName(
+              sheetNames[cmd.sheetId],
+              Object.values(sheetNames)
+            );
```

Comment:
getDuplicateSheetName uses translations, so we'll end up with the exact same issue with different users having different sheet names after the repair :see_no_evil: